implementation/jobberknoll/core/domain/entities/account.ts

```diff
@@ -0,0 +1,29 @@
+import type { UUID } from "@jobberknoll/core/shared";
+
+type AccountBase = {
+  id: UUID;
+  fullName: string;
+  email: string;
+  hashedPassword: string;
+  isActive: boolean;
+  lastModified: number;
+};
+
+export type Passenger = AccountBase & {
+  type: "passenger";
+  phoneNumber?: string;
+};
+
+export type Driver = AccountBase & {
+  type: "driver";
+};
+
+export type Admin = AccountBase & {
+  type: "admin";
+};
+
+export type Inspector = AccountBase & {
+  type: "inspector";
+};
+
+export type Account = Passenger | Driver | Admin | Inspector;
```

Comment:
This somewhat relates to your comment at my place. In my opinion, we should add a `Guest` here. But we should discuss this in a larger group.

Reply:
It's a domain type meant to be used for business logic and later the persistence/data-access layer is built upon it. So I don't see the need to hold `Guest` here, since guests don't exist in the database (and neither in the app state, since our services should be stateless).

I'm guessing you thought it was a role enumeration? As in:
```
type Role = "guest" | "passenger" | "driver" | "admin" | "inspector";
```

I don't have such type yet, but if I will, it would contain a `guest` literal 👍